### DISCUSSION TITLE
[5.x] FR: Entry type re-labelling

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#14779] [Ideas] [answered]
Similarly to how custom fields can be re-labelled in field layouts, it'd be great if it was possible to relabel entry type names and handles in (or via) the entry types selector for sections, Matrix and CKEditor fields.

In Craft 5, entry type handles and names are required to be unique, which mean that similar entry types will have to be given more specific – and not necessarily _client-/author-friendly_ – names, e.g. "Text with all options", "Images (limited to 5)" etc.

Also, in lieu of any organisational features for entry types, I think a lot of developers will opt to give entry types contextual names like "News Block", "Image Partial" or "Blog Page Entry", in order to achieve some kind of organisation for them inside the control panel. IMO, none of these are excellent names to present to authors in the UI, and it would be great if it was possible to relabel them to simply "News", "Image" and "Blog Page", for the different context(s) where they are used.

In terms of UI, potentially a relabel feature could be implemented by turning the entry type selector component into something a little bit more like a mini field layout designer? This could provide the prerequisite UI for implementing additional per-context entry type features like per-field entry type groups, or even conditional entry types.

Related: 

https://github.com/craftcms/cms/issues/14774
https://github.com/craftcms/cms/discussions/14726
https://github.com/craftcms/cms/discussions/13307
https://github.com/craftcms/cms/discussions/14179#discussioncomment-8312309

**Comment by brandonkelly** [comment]
This thought crossed my mind when reading #14774. I could see us adding them when we get around to adding conditions to entry types, as you mentioned.

In the meantime, you can technically pull of multiple entry types with the same name using static translations (see https://github.com/craftcms/cms/issues/14774#issuecomment-2046127930).

**Comment by Anubarak** [comment]
@brandonkelly Not sure if that belongs here or in another discussion: have you thought about adding the handle to the entry type select?
We had trouble creating larger projects with about 60 different entry types (many sections, many matrix blocks, "categories" are moved to entry types etc) to differentiate between the correct types very well.

When you have a "matrix block" for content modelling that should display the latest 3 news of a news section together with some kind of News Index entry type  you'll end up with 3 entry types that are all named "News" (or something like that).
Adding the handle to the entry type select will make our work much easier - especially when you add a real re-label feature.

We can add our custom naming convention for handles and see the purpose of the type via handle name `newsIndex`, `newsBlock`, `news` that will all become just `News` in the list, when we translate it. So right now we have to trial and error the correct entry types until we find the correct one in the select sometimes.

**Reply by mmikkel** [reply]
Adding entry type handles to the entry type selector component (similarly to how field names + handles are visible in field layout designers) would mitigate the problem that the unique name requirement solves. 

Personally, I think showing handles in most developer facing views is a good idea, regardless. However, I also think that Craft 5 encouraging/forcing unique names for entry types is actually a good call (if it were feasible to retroactively implement at this point, I think field names should also be required to be unique) – so re-labelling would be the ideal solution IMO.

**Comment by brandonkelly** [comment, marked as answer]
Added support for overriding entry type names & handles for Craft 5.6 🎉 (#16453)

**Reply by brandonkelly** [reply]
Craft 5.6.0 is out now 🚀
